### PR TITLE
per-user platform limit of concurrent syncs

### LINKED ISSUE #4395: There should be a per-user platform limit of concurrent syncs
@danicarrion raised an issue we have: There is a limit of concurrent imports per user, but there is no limit of concurrent syncs per user. So I can create 50 syncs almost at once, and they would eat all cloud workers for themselves, not allowing anybody to import until they finish.

We should implement a platform limit so only up to X (I'd say 3, separate from the imports limit), so only up to X get enqueued, and when one finishes next sweep will have available slots again (now sweeps are done each 2-3 mins so is not problematic).

I estimate ~4h of time to build and (mostly manually) test it, as the PlatfomLimits system is already there and easy to extend.

@saleiva @javisantana

### PR DESCRIPTION
**Pull request by Kartones**
Closes #4395

### CODE REVIEW

**Comment by Kartones** [comment]
@juanignaciosl CR please

File: services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+        attr_accessor :redis, :expire_ttl
+
+        def subkey
+          'Importer:UserConcurrentSyncsAmount'
+        end
+
+        # @param context mixed
+        # @return bool
+        def is_over_limit(context)
```
Comment (juanignaciosl) [inline services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb]: Shouldn't this be `is_over_limit?`, with quotation mark?

**Comment by juanignaciosl** [comment]
After the comment, LGTM.

File: services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb (hunk 1 above)
Comment (Kartones) [inline services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb]: Check `abstract_limit`, this is the internal implementation (which varies on each limit), AbstractLimit is the one that provides `is_over_limit?` and `is_over_limit!`.

Comment (juanignaciosl) [inline services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb]: ah, ok!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5444/))

File: services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb (hunk 1 above)
Comment (Kartones) [inline services/platform-limits/lib/importer/user_concurrent_syncs_amount.rb]: https://github.com/CartoDB/cartodb/blob/master/services/platform-limits/lib/abstract_limit.rb#L45

That's also why has `protected` scope, so outside can only use `xxx?` and `xxx!` versions

**Pull request merged**